src/array/flattenDeep.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Flattens `array` recursively.
+ * @param {Array} array
+ * @returns {Array} Returns new flattened array
+ */
+function flattenDeep(array) {  
+  let flattenedArray = []
```

Comment:
I don't want to restrict the variable "flattendArray" scope to the function only for that created a inner /child function (closure). @pravinpareek @hemantkumargaikwad What do you think?

Reply:
Yes, it seems good.